### ISSUE TITLE
question: is there any short term roadmap towards flox-containerize

### ISSUE THREAD

**Issue by networkhermit** [flox/flox#14]
hi, I love the idea of packaging containers in nixpkgs. I'm wondering if you have any short term roadmap to improve flox-containerize.

I have the following feedback and questions:

1. Currently there seems no way for `flox develop` and `flox containerize` to distinguish dev dependencies from runtime dependencies.
2. Is it possible to create container image that not depend on bash or shell? Like using static compilation in go: https://docs.docker.com/build/building/base-images/#create-a-simple-parent-image-using-scratch
3. Does the lines in [flox catalog](https://floxdev.com/docs/reference/concepts/catalog/#catalog) suggest the package version are only limited to the latest 9 ones? If that's the case then how can I ensure the flox environments keeps working after a long time and still guarantee binary cache hit?
4. Do you have plans to support [muti-platform images](https://docs.docker.com/build/building/multi-platform/)?
5. I know maybe it's out of scope, but have you considered adding vm or firecracker images to flox ecosystem in the future?

**Comment by aakropotkin** [comment]
These are great questions. I've directed them over to a few team members to answer and once we can organize answers for you'll I'll report back.

**Comment by networkhermit** [comment]
@aakropotkin 

Thank you for the detailed clarification.

> I'm curious - from your perspective what would be the advantage/impact of having a container built from "scratch" in this way?

It's more about minimal dependencies/image size, and the capability. It's a direct comparison from what docker supports. BTW, in some constrained environments image size does matter.

> Is your request to automatically render containers capable of supporting multiple system types, or do you specifically require those to be a single file (along with the size implications thereof)? On our side we are working on a feature to automatically render named environments for multiple system types but haven't been looking into multi-platform images, primarily on account of the size of the resulting images.

docker support building multi-platform images, as the OS/ARCH showed in [dockerhub hello-world](https://hub.docker.com/layers/library/hello-world/latest/images/sha256-7e9b6e7ba2842c91cf49f3e214d04a7a496f8214356f41d81a6e6dcad11f11e3?context=explore), it's actually a list of images. I see this topic is better tracked in [#82](https://github.com/flox/flox/issues/82).

P.S. Is it possible to use nixos unstable directly instead floxpkgs channels when using flox?

**Comment by aakropotkin** [comment]
Yeah so you can upstream nixpkgs by providing installable URIs such as `flox install github:NixOS/nixpkgs#hello;` or writing a `flox.nix` file with:
```nix
{
  packages."github:NixOS/nixpkgs".hello = {};
}
```
